mongo_repository.go

```diff
@@ -288,6 +288,38 @@ func (r *MongoRepository[T]) lastSnapshot(ctx context.Context, objectID string)
 	return &lastSnapshot, nil
 }
 
+func (r *MongoRepository[T]) Remove(ctx context.Context, objectID string, object T) (T, error) {
+	if exists, err := r.Exists(ctx, objectID); err != nil || !exists {
+		return object, errors.New("cannot load unknown object")
+	}
+	err := r.Load(ctx, objectID, object)
+	if err != nil {
+		return object, err
+	}
+
+	_, err = r.collection.DeleteMany(ctx, bson.D{{"objectid", objectID}})
+	if err != nil {
+		return object, err
+	}
+
+	event := NewEvent(objectID, "removed", object.LastVersion(), []byte{})
+	events := []Event{event}
+	records := toRecords(events)
+	_, err = r.collection.InsertMany(ctx, records)
```

Comment:
C'est probablement une bonne idée de laisser une trace dans l'event store des objets supprimés, cependant ce n'est pas le cas dans l'implémentation de PythonDDD (Just so you know)

Il faut aussi prendre en compte que si on reprojette un event store avec des elements supprimés, alors les listeners de l'event `removed` seront appelés sur données inexistantes. En théorie ça ne pose pas de problème, en pratique peut-être?

Je serais quand même d'avis de **conserver** cet event, ça pourrait servir

Reply:
Je suis au courant pour PythonDDD oui, j'y ai pas mal chipoté ;). 

Je suis d'accord pour l'aspect re-projection. Mais dans mon esprit, une suppression sur une projection devrait être idempotente. 

Pour ce qui est de le conserver, je suis d'accord aussi. C'est super utile pour savoir si une chose qu'on ne voit plus dans l'état actuel des choses a en fait existé à un moment donné !

Reply:
Gardons ainsi alors :)